addons/web_editor/controllers/main.py

```diff
@@ -63,8 +63,13 @@ def get_existing_attachment(IrAttachment, vals):
     so, None otherwise.
     """
     fields = dict(vals)
-    # Falsy res_id defaults to 0 on attachment creation.
-    fields['res_id'] = fields.get('res_id') or 0
+    if fields['res_model'] == 'web_editor.media':
+        # For a media attachment, the res_id is the media's id, which (if it
+        # exists) we don't know at this point. We do the comparison on the rest.
+        fields.pop('res_id', None)
```

Comment:
`get_existing_attachment` checks the attachment itself, not the media (as we're in the middle of trying to upload an image, we only have the data of the file itself). So the `res_id` that is passed here as a value is the id of the current record in which the attachment is inserted.
But as said earlier, the "true" `res_id` is stored on the media record, not the attachment anymore. So if I don't do this here, it will never find any existing attachment, because as long as I don't know if an attachment exists, I can't find the related media record, so I can't access the `res_id` that is checked here.

I was about to say that "if I can get rid of all routes that create basic attachments in `web_editor`, that condition wouldn't have to be an exception anymore and I could just remove the `res_id` from the comparison", but the function is at least used in `modify_image()` which doesn't create media records (and from my understanding I don't think it should?).

Reply:
Ok I see. Then what I don't see is the use we have for `get_existing_attachment` at all when `fields['res_model'] == 'web_editor.media'`. Maybe it should not be called at all? :open_mouth:

Maybe it is to prevent uploading the same image twice in the media dialog for the same model+id+field? :open_mouth: Did not know we had that sort of check... do we? If this is it: is not this update breaking something? If you prevent creation of duplicate for all related-to-media attachments without considering what the web_editor.media are themselves attached to? Like you could not upload the same image on different products maybe?

=> In any case, there is something strange here.

Reply:
> Maybe it is to prevent uploading the same image twice in the media dialog for the same model+id+field?

Yes that's the point, and yes you were right, that created some bugs (the same image could be used on 2 different models, but then couldn't be retrieved on the 2nd one). I'll add a check about that.